// gitbook-plugin-livere/_assets/plugin.js
(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.GitBookPlugin = f()}})(function(){var define,module,exports;return (function(){function r(e,n,t){function o(i,f){if(!n[i]){if(!e[i]){var c="function"==typeof require&&require;if(!f&&c)return c(i,!0);if(u)return u(i,!0);var a=new Error("Cannot find module '"+i+"'");throw a.code="MODULE_NOT_FOUND",a}var p=n[i]={exports:{}};e[i][0].call(p.exports,function(r){var n=e[i][1][r];return o(n||r)},p,p.exports,r,e,n,t)}return n[i].exports}for(var u="function"==typeof require&&require,i=0;i<t.length;i++)o(t[i]);return o}return r})()({1:[function(require,module,exports){
"use strict";

require(["gitbook", "jQuery"], function (gitbook, $) {
  var livere_uid;
  var timer;

  function loadInstance() {
    if (!window) {
      return;
    }
    if (window.document.getElementById("livere-js") === null) {
      var script = window.document.createElement("script");
      script.async = true;
      script.src = "https://cdn-city.livere.com/js/embed.dist.js";
      script.id = "livere-js";
      script.onload = function () {
        setTimeout(function () {
          clearTimeout(timer);
          timer = null;
        }, 300);
      };
      window.document.body.appendChild(script);
      return;
    }
    if (window.LivereTower && timer === null) {
      timer = setTimeout(function () {
        try {
          window.LivereTower.reload();
        } catch (e) {
          createLivereDiv();
          window.LivereTower.init();
        }
        clearTimeout(timer);
        timer = null;
      }, 300);
    }
  }

  function createLivereDiv() {
    var $livereDiv = $("<div>", {
      id: "lv-container",
      "data-id": "city",
      "data-uid": livere_uid
    });
    $(".book-body .page-inner").append($livereDiv);
  }

  gitbook.events.bind("start", function (e, config) {
    config.livere = config.livere || {};
    livere_uid = config.livere.uid;
    loadInstance();
    createLivereDiv();
  });

  gitbook.events.bind("page.change", loadInstance);
});

},{}]},{},[1])(1)
});
